src/service/viewer-impl.js

```diff
@@ -860,12 +860,33 @@ export class Viewer {
     if (!this.hasCapability('fragment')) {
       return Promise.resolve('');
     }
-    return this.sendMessageUnreliable_('fragment', undefined, true).then(
+    return this.sendMessageUnreliable_('getFragment', undefined, true).then(
       hash => hash || ''
     );
   }
 
   /**
+   * Update the fragment of the viewer if embedded in a viewer,
+   * otherwise update the page url fragment
+   * @param {string} fragment
+   * @return {!Promise}
+   */
+  updateFragment(fragment) {
+    if (!this.isEmbedded_) {
+      if (this.win.history.replaceState) {
+        const newUrl = this.win.location.href.split('#')[0] + '#' + fragment;
+        this.win.history.replaceState({}, '', newUrl);
+      }
+      return Promise.resolve();
+    }
+    if (!this.hasCapability('fragment')) {
+      return Promise.resolve();
+    }
+    return /** @type {!Promise} */ (this.sendMessageUnreliable_(
```

Comment:
When I send the message to viewer to let it update the fragment, the viewer should update the google.com url fragment with the `fragment` we provided. When amp page is embedded in search viewer, we want the share-tracking fragment update to be reflected in the google search url. Search viewer should do something like `win.history.replaceState` on their side.